### ISSUE TITLE
axios timeout s not respected for chunked requests

### ISSUE THREAD

**Issue by n0th1ng-else** [axios/axios#5886]
### Describe the bug

Hello dear team. hope you are doing well.

I am about to report a bug, although it probably works as expected, then I would ask you to clarify this behavior in the documentation.

regarding the axios timeout - it works fine for regular requests. But it is not respected for chunked requests. Imagine the server is streaming the response chunks for 30 sec and I would expect setting 10 sec timeout will abort execution. Instead, it keeps hanging and waiting on the last chunk, so the request can be finished.

It is quite non-intuitive and confusing and I wonder if it was designed like that, or you consider it as a bug and we want to fix it.
I have attached the github repository with a reproduction example.  https://github.com/n0th1ng-else/axios-streaming-timeout

could you please take a look?

<img width="642" alt="image" src="https://github.com/axios/axios/assets/46719800/6fcd0990-c25d-4300-a7ca-4976377f24aa">

### To Reproduce

https://github.com/n0th1ng-else/axios-streaming-timeout

### Expected behavior

The request is aborted after 10 sec due to timeout, even tho it is half way through

### Axios Version

1.5.0

### Browser

NodeJS express + axios

### Node.js Version

16.16.1

### OS

MacOS Venture 13.4.1

**Comment by SagarKapoorin** [comment]
Hi there,
I'm relatively new to this project, so please correct me if I'm mistaken.
i have read your commits (file )and come with idea that:-
Axios may not forcefully terminate the request while it's still expecting more data to come.So,i think in that case timeout function will not work rather you . We have to make a separate timeout function in addition to axios's timeout function.
By implementing this custom timeout mechanism, you should have better control over requests with chunked responses and ensure that the timeout is respected

**Comment by n0th1ng-else** [comment]
Yes exactly, once the first chunk is received, Axios drops the timeout and considers the request answered in time. This I think raises the debate on at what moment we consider the request timed out - at the first byte received or the last byte received. 

I found a workaround this issue (the issue in my case, as it generally looks like designed this way), the solution is to use AbortSignal. It cancels the request regardless, see the example:

```javascript
	const AXIOS_TIMEOUT = 5000;

   axios.request({
        method: "POST",
        url,
        timeout: AXIOS_TIMEOUT,
		signal: AbortSignal.timeout(AXIOS_TIMEOUT), // Ensures the request will be cancelled
    })
```
More about AbortSignal https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal/timeout_static

More about how axios uses it https://axios-http.com/docs/cancellation

I would still love to clarify this behavior in the documentation

**Comment by jasonsaayman** [comment]
yeah as @n0th1ng-else has pointed out correctly that i how axios works
